knime_extension/src/nodes/spatialnetwork.py

```diff
@@ -1423,3 +1423,200 @@ def execute(self, exec_context: knext.ExecutionContext, input1, input2):
+        Note that information on historic road speeds is always used.""",
+        default_value=True,
+    )
+
+    route_type = knext.EnumParameter(
+        "Route type",
+        """Determines the type of route used for the isochrone calculation. 
+        Different route types can result in different isochrones.""",
+        default_value=_TomTomRouteType.get_default().name,
+        enum=_TomTomRouteType,
+    )
+
+    travel_mode = knext.EnumParameter(
+        "Travel mode",
+        """Specifies the mode of travel for the isochrone calculation, which can 
+        significantly impact the shape and extent of the isochrone.""",
+        default_value=_TomTomTravelMode.get_default().name,
+        enum=_TomTomTravelMode,
+    )
+
+    tomtom_api_key = knext.StringParameter(
+        "TomTom API Key",
+        """The 
+        [TomTom API key](https://developer.tomtom.com/knowledgebase/platform/articles/how-to-get-an-tomtom-api-key/)
+        is required to authenticate requests to the 
+        [Calculate Reachable Range service](https://developer.tomtom.com/routing-api/documentation/routing/calculate-reachable-range)
+        which is part of the [Routing API](https://developer.tomtom.com/routing-api/documentation/routing/routing-service)
+        provided by TomTom. 
+        To get an API key, click
+        [here.](https://developer.tomtom.com/knowledgebase/platform/articles/how-to-get-an-tomtom-api-key/)
+        For details about the pricing go to the [TomTom pricing page.](https://developer.tomtom.com/store/maps-api)""",
+        default_value="",
+        # TODO: Validate that the API key is present
+    )
+
+    timeout = knext.IntParameter(
+        "Request timeout in seconds",
+        "The maximum time in seconds to wait for the request to the TomTom API to succeed.",
+        120,
+        min_value=1,
+        is_advanced=True,
+    )
+
+    _COL_GEOMETRY = "Geometry"
+    _COL_ISOCHRONE = "Isochrone"
+
+    def configure(self, configure_context, input_schema_1):
+        self.c_geo_col = knut.column_exists_or_preset(
+            configure_context, self.c_geo_col, input_schema_1, knut.is_geo
+        )
+        # TODO: Return result schema with the origin id and bounds column
+
+        return None
+
+    def execute(self, exec_context: knext.ExecutionContext, input1):
+        tomtom_base_url = "https://api.tomtom.com/routing/1/calculateReachableRange/"
+        # a sample url:
+        # url = """https://api.tomtom.com/routing/1/calculateReachableRange/52.50931,13.42936/json?energyBudgetInkWh=43&avoid=unpavedRoads&vehicleEngineType=electric&constantSpeedConsumptionInkWhPerHundredkm=50,8.2:130,21.3&key=%s"""%tomkey
+        import requests
+        import json
+        from shapely.geometry import Polygon
+
+        c_gdf = knut.load_geo_data_frame(input1, self.c_geo_col, exec_context)
+
+        # TODO: Execute in a loop with progress and cancellation to support a table with multiple input points
```

Comment:
See todo

Reply:
The TODO is no longer valid. However we need to compute the global centroid for multiple inputs like in the Road Network Isochrone Map.  In addition we need to support multiple budgets.